modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -7122,6 +7122,7 @@ public void setRemoveEnumValuePrefix(final boolean removeEnumValuePrefix) {
     public void addOneOfNameExtension(ComposedSchema s, String name) {
         if (s.getOneOf() != null && s.getOneOf().size() > 0) {
             s.addExtension("x-one-of-name", name);
+            s.setType(name);
```

Comment:
This is all the way in default code-gen - does this affect more than just Java/Spring?